src/OpenLoco/src/GameCommands/CreateTown.cpp

```diff
@@ -0,0 +1,200 @@
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::World;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00496FE7
+    // TODO: only used here, but move to town manager?
+    static Town* initialiseTown(Pos2 pos)
+    {
+        registers regs;
+        regs.eax = pos.x;
+        regs.ecx = pos.y;
+        call(0x00496FE7, regs);
+
+        if (regs.esi != -1)
+            return reinterpret_cast<Town*>(regs.esi);
+        else
+            return nullptr;
+    }
+
+    // 0x00496E09
+    static bool checkSurroundings(Pos2 pos, bool checkSurroundingWater)
+    {
+        auto tile = TileManager::get(pos);
+        auto* surfaceEl = tile.surface();
+        if (surfaceEl->slope() != 0 || surfaceEl->water() != 0)
+        {
+            return false;
+        }
+
+        if (checkSurroundingWater)
+        {
+            auto* landObj = ObjectManager::get<LandObject>(surfaceEl->terrain());
+            if (landObj->hasFlags(LandObjectFlags::isDesert | LandObjectFlags::noTrees))
+            {
+                auto nearbyWaterTiles = TileManager::countNearbyWaterTiles(pos);
+                if (nearbyWaterTiles < 10 && getGameState().rng.randNext() & 0xFF)
+                {
+                    return false;
+                }
+            }
+        }
+
+        for (auto& town : TownManager::towns())
+        {
+            auto manhattanDistance = Math::Vector::manhattanDistance(pos, Pos2(town.x, town.y));
+            if (manhattanDistance < 768)
+            {
+                setErrorText(StringIds::too_close_to_another_town);
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    // 0x00496C22
+    static uint32_t createTown(const TownPlacementArgs& args, const uint8_t flags)
+    {
+        auto& gameState = getGameState();
+        Pos2 pos = args.pos;
+        Town* newTown = nullptr;
+
+        if (args.pos.x == -1)
+        {
+            bool foundPos = false;
+            for (auto attempts = 200; attempts > 0; attempts--)
+            {
+                uint32_t rand = gameState.rng.randNext();
+                auto tilePos = TilePos2(((rand >> 16) * kMapColumns) >> 16, ((rand & 0xFFFF) * kMapRows) >> 16);
+                Pos2 attemptPos = toWorldSpace(tilePos);
+
+                if (attemptPos.x < 384 || attemptPos.y < 384 || attemptPos.x > 11904 || attemptPos.y > 11904)
+                {
+                    continue;
+                }
+
+                if (checkSurroundings(attemptPos, true))
+                {
+                    pos = attemptPos;
+                    foundPos = true;
+                    break;
+                }
+            }
+
+            if (!foundPos)
+            {
+                return FAILURE;
+            }
+        }
+        else
+        {
+            bool foundPos = false;
+            for (int attempts = 0; attempts < 40; attempts++)
+            {
+                Pos2 attemptPos = pos;
+                if (attempts <= 10)
```

Comment:
0 is <= 10. Think you'll need a `if (attempts == 0){ // do nothing attemptPos = pos }`